scalameta/io/shared/src/main/scala/scala/meta/io/AbsolutePath.scala

```diff
@@ -1,12 +1,18 @@
-package scala.meta.io
+package scala.meta
+package io
 
 import java.io._
+import java.net._
 import org.scalameta.data._
 import scala.meta.internal.io.{FileIO, PathIO}
 
 @data class AbsolutePath private (value: String) {
-  override def toString: String = value
+  def syntax: String = value
+  def structure: String = s"""AbsolutePath("$value")"""
+  override def toString: String = syntax
```

Comment:
I think that all our data structures should have pairs of `syntax` and `structure` methods.

In this pull request, this is partially implemented in a really adhoc manner. In the future, I suggest that we enforce this with `@data` and `@leaf` annotations. Moreover, we can use these annotations to easily derive `def structure: String`. That'll save quite some boilerplate.